src/Core/System/SalesChannel/Context/BaseSalesChannelContextFactory.php

```diff
@@ -279,19 +286,26 @@ private function getCashRounding(CurrencyEntity $currency, ShippingLocation $shi
         return [$currency->getItemRounding(), $currency->getTotalRounding()];
     }
 
-    private function getLanguageInfo(?LanguageCollection $languages, string $currentLanguageId): LanguageInfo
+    private function getLanguageInfo(Context $context): LanguageInfo
     {
-        $currentLanguage = $languages?->get($currentLanguageId);
-        if ($currentLanguage === null) {
+        $currentLanguageId = $context->getLanguageId();
+        $criteria = (new Criteria([$currentLanguageId]))->addFields([
+            'name',
+            'translationCode.code',
+            'locale.code',
+        ]);
+
+        $currentLanguage = $this->languageRepository->search($criteria, $context)->getEntities()->get($currentLanguageId);
+        if (!$currentLanguage instanceof PartialEntity) {
             throw SalesChannelException::languageNotFound($currentLanguageId);
         }
 
-        $locale = $currentLanguage->getTranslationCode() ?? $currentLanguage->getLocale();
-        \assert($locale !== null, 'At least the localeId is required, so the fallback should never be null');
+        $locale = $currentLanguage->get('translationCode') ?? $currentLanguage->get('locale');
+        \assert($locale instanceof PartialEntity, 'At least the localeId is required, so the fallback should never be null');
 
         return new LanguageInfo(
-            $currentLanguage->getTranslation('name') ?? $currentLanguage->getName(),
-            $locale->getCode(),
+            $currentLanguage->get('name'),
+            $locale->get('code'),
         );
```

Comment:
For now, this is a performance penalty, since you're doing an additional repository call in 6.7, since the association is still loaded anyways. Couldn't we just do a fallback for 6.7 and only do the call with `v6.8.0.0` enabled?